### PR TITLE
Removes a trycatch and a null check in bodypart ashing

### PR DESCRIPTION
**Pull request by Aranclanos**
### Purpose
Removes the CannotBeAshed bool, ashing will now check for fire resistance.
Organizes bodyPart damage, organ damage and trauma calls.
Removes unneeded an unneeded try catch in body part ashing.
Adds DamageOrgans() to differentiate, separate and organize the damage inbetween a body part and an organ.

### CODE REVIEW

File: UnityProject/Assets/Scripts/HealthV2/Living/Damage/BodyPartDamage.cs [hunk 1]
```diff
@@ -221,52 +224,67 @@ public void AffectDamage(float damage, int damageType)
...
+				}
 			}
 
-			if(Severity <= DamageSeverity.LightModerate) return;
-			if (damageType == DamageType.Burn || attackType == AttackType.Fire
-			                                  || attackType == AttackType.Laser || attackType == AttackType.Energy)
+			if (damageType == DamageType.Burn || attackType == AttackType.Fire ||
+			    attackType == AttackType.Laser || attackType == AttackType.Energy)
 			{
-				ApplyTraumaDamage(TraumaticDamageTypes.BURN, true);
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/HealthV2/Living/Damage/BodyPartDamage.cs:266]: why was this pass removed?

Comment (Aranclanos) [reply UnityProject/Assets/Scripts/HealthV2/Living/Damage/BodyPartDamage.cs:266]: it never did anything due to the `if(Severity <= DamageSeverity.LightModerate) return;` line that I also removed a couple lines before
so now the lightmoderate check will still exist and used, but inside ApplyTraumaDamage

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/HealthV2/Living/Damage/BodyPartDamage.cs:266]: Everything seems to be good, tested as well and no issues creeping up.

**Review by MaxIsJoe — APPROVED** [review]
Everything seems to be good, tested as well and no issues creeping up.

**Review by NoooneyDude — APPROVED** [review]

File: UnityProject/Assets/Scripts/HealthV2/Living/LivingHealthMasterBase.cs [hunk 2]
```diff
@@ -287,13 +284,13 @@ private void PeriodicUpdate()
 		{
 			if (IsDead == false)
 			{
-				for (int i = BodyPartList.Count - 1; i >= 0; i--)
+				foreach (var bodyPart in BodyPartList)
 				{
-					BodyPartList[i].ImplantPeriodicUpdate();
+					bodyPart.ImplantPeriodicUpdate();
 				}
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/HealthV2/Living/LivingHealthMasterBase.cs:290]: Are you sure these for loops arent needed due to changing the BodyPartList and so needs to be looped backwards?

Comment (Aranclanos) [reply UnityProject/Assets/Scripts/HealthV2/Living/LivingHealthMasterBase.cs:290]: yes

**Review by ThatDan123 — APPROVED** [review]

**Pull request merged**